lib/components/util/formatted-validation-error.js

```diff
@@ -0,0 +1,31 @@
+import { FormattedMessage } from 'react-intl'
+import PropTypes from 'prop-types'
+
+/**
+ * Returns a FormattedMessage component for a yup validation error in order to
+ * keep react-intl message IDs hardcoded.
+ */
+export default function FormattedValidationError ({type}) {
+  switch (type) {
```

Comment:
Here again, I would use template strings to generate the ID and PropTypes (https://stackoverflow.com/a/60912152) to validate the input being correct.